### PR TITLE
Added config store to historians

### LINKED ISSUE #1354: Allow ForwardHistorian to be "paused"
During upgrade there is a slice of time where by the databases on a central won't be available, however the platform will be available.  If the forward historian could just start caching rather than publishing to the second bus, it would ensure that data was captured during the upgrade.

Note this is related to requiring peers to be present before publishing.

### LINKED ISSUE #1484: Spurious error when using config edit to create a new config
RPC error is thrown, but new config entry is created and can then be edited. Error just being thrown on a check and could cause confusion.

```` 
('RPC ERROR', 'Traceback (most recent call last):
File "/home/jer/git/volttron-craig/volttron/platform/vip/agent/subsystems/rpc.py", line 169, in method    
  return method(*args, **kwargs)
File "/home/jer/git/volttron-craig/volttron/platform/store.py", line 256, in manage_get_metadata
  raise KeyError(\'No configuration file "{}" for VIP IDENTIY {}\'.format(config_name, identity))
  KeyError: \'No configuration file "config" for VIP IDENTIY forwarder2\'\n')
`````

### PR DESCRIPTION
**Pull request by craig8**
Fixes #1354 - pause forwarder
Fixes #1484, #1485, #1400 83

Added methods for creating and validating publishes of device messages for tests.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/VOLTTRON/volttron/pull/1494%23discussion_r141445871%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1494%23discussion_r141446729%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1494%23discussion_r141447610%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20efb914fd3040ccb413050e76b719144393e145fb%20services/core/ForwardHistorian/README.rst%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1494%23discussion_r141445871%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Consider%20adding%20alerts%20as%20well%22%2C%20%22created_at%22%3A%20%222017-09-27T19%3A35%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/6992449%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/jhaack%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/ForwardHistorian/README.rst%3AL83-90%22%7D%2C%20%22Pull%20efb914fd3040ccb413050e76b719144393e145fb%20volttron/platform/agent/base_historian.py%20206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1494%23discussion_r141447610%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Put%20a%20link%20to%20the%20project%20this%20code%20is%20borrowed%20from%22%2C%20%22created_at%22%3A%20%222017-09-27T19%3A42%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/6992449%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/jhaack%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20volttron/platform/agent/base_historian.py%3AL1375-1404%22%7D%2C%20%22Pull%20efb914fd3040ccb413050e76b719144393e145fb%20services/core/ForwardHistorian/config%202%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1494%23discussion_r141446729%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Remove%20ipc%20and%20default%20to%20localhost%22%2C%20%22created_at%22%3A%20%222017-09-27T19%3A38%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/6992449%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/jhaack%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/ForwardHistorian/config%3AL1-5%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull efb914fd3040ccb413050e76b719144393e145fb services/core/ForwardHistorian/README.rst 5'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1494#discussion_r141445871'>File: services/core/ForwardHistorian/README.rst:L83-90</a></b>
- <a href='https://github.com/jhaack'></a> Consider adding alerts as well
- [x] <a href='#crh-comment-Pull efb914fd3040ccb413050e76b719144393e145fb services/core/ForwardHistorian/config 2'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1494#discussion_r141446729'>File: services/core/ForwardHistorian/config:L1-5</a></b>
- <a href='https://github.com/jhaack'></a> Remove ipc and default to localhost
- [x] <a href='#crh-comment-Pull efb914fd3040ccb413050e76b719144393e145fb volttron/platform/agent/base_historian.py 206'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1494#discussion_r141447610'>File: volttron/platform/agent/base_historian.py:L1375-1404</a></b>
- <a href='https://github.com/jhaack'></a> Put a link to the project this code is borrowed from

<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1494?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1494?approve=1'></a>&nbsp;<a href='https://github.com/VOLTTRON/volttron/pull/1494'></a>
<a href='#crh-end'></a>

---
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/1494)

***
***
***

### CODE REVIEW

**Comment by craig8** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/VOLTTRON/volttron/pull/1494'>Refresh</a>.<a href='#crh-update-none'></a>

File: services/core/ForwardHistorian/README.rst [hunk 1]
```diff
@@ -83,7 +83,7 @@ by the ForwardHistorian agent.
         #   Unlike other historians, the forward historian can re-publish from
         #   any topic.  The custom_topic_list is prefixes to subscribe to on
         #   the local bus and forward to the destination instance.
-        "custom_topic_list": [],
+        "custom_topic_list": ["actuator"],
```
Comment (jhaack) [inline services/core/ForwardHistorian/README.rst]: Consider adding alerts as well

File: services/core/ForwardHistorian/config [hunk 2]
```diff
@@ -1,12 +1,4 @@
 {
     "destination-vip": "ipc://@/home/volttron/.volttron/run/vip.socket",
```
Comment (jhaack) [inline services/core/ForwardHistorian/config]: Remove ipc and default to localhost

File: volttron/platform/agent/base_historian.py [hunk 3]
```diff
@@ -1348,6 +1375,29 @@ def _setupdb(self):
         self._connection.commit()
 
 
+#Shamelessly borrowed from gsqlite.
```
Comment (jhaack) [inline volttron/platform/agent/base_historian.py]: Put a link to the project this code is borrowed from

**Pull request merged**